### ISSUE TITLE
Random schematic decor

### ISSUE THREAD

**Issue by tenplus1** [minetest/minetest#4485]
When registering simple biome decoration we can add things like:
```
decoration = {"flowers:rose", "fowers:tulip"}
```
... and have it choose randomly between the two when placing, but... why can't we do this for schematics also ?  This would be a great addition that would save you registering 2x different schematic decorations and squashing them together inside a single biome, instead giving the option to those randomly between schems in a table e.g.:
```
schematic = {"mymod:schem1", "mymod:schem2"}
```
... and having schematic decoration work with spawn_by would be amazing also :)

**Comment by paramat** [comment]
Spawn by is going ahead.

**Comment by tenplus1** [comment]
w00t!

**Comment by tenplus1** [comment]
@paramat - will random schematic selection be looked at for mapgens ?

**Comment by paramat** [comment]
Good idea but i don't know how to code it.

**Comment by MarkuBu** [comment]
Didn't know that there was already an issue with this idea. I want to suggest an optional chance parameter. I quote from my issue

> As option a chance would be great. Each schematic can have a chance how often this schematic will be placed. I would propose, the higher the chance value, the lower the chance to place a schematic. This makes it easier to define rare schematics within the common once (chance = 1 could be default)
> 
> Example:
> 
> ```
> minetest.register_decoration({
> 	deco_type = "schematic",
>  	place_on = {"default:dirt_with_grass"},
> 	sidelen = 16,
> 	noise_params = {
> 		offset = 0.01,
> 		scale = 0.001,
>  		spread = {x = 250, y = 250, z = 250},
> 		seed = 2,
> 		octaves = 3,
> 		persist = 0.66
> 	},
> 	biomes = {"mixed_forest"},
> 	y_min = 10,
> 	y_max = 80,
> 	schematic = {
> 		{schem = "appletree"},
> 		{schem = "aspentree"},
> 		{schem = "pinetree"},
> 		{schem = "oldpinetree", chance = 10}
> 		}
> 	flags = "place_center_x, place_center_z",
> })
> 
> ```
>

**Comment by sofar** [comment]
Just a thought, I think it could be much more simple. The current documentation reads:

`schematic = "foobar.mts",`

What we could do is allow this to do the same thing as sound files:

`schematic = "foobar.mts",` -- old style, will only ever load "foobar.mts"
`schematic = "foobar",` -- new style, will load "foobar.1.mts", "foobar.2.mts" etc. and pick one randomly

That doesn't give you the chance parameter, which could be interesting for sure.

**Comment by paramat** [comment]
I don't think that would be possible as that the current `schematic =` parameter accepts a Lua table or a path to a file `schematic = minetest.get_modpath("default") .. "/schematics/acacia_bush.mts",` so perhaps the best format is a table of Lua tables or a table of paths. This would be consistent with how multiple choice simple decorations work with a table of nodes.

**Comment by MarkuBu** [comment]
What about lua table schematics?

**Comment by MarkuBu** [comment]
> I would rather not add a chance parameter, that could be simulated by having multiple identical entries if really wanted.

That would mean that the same schematic is loaded to memory multiple times and my example would look like this

```
	schematic = {
		schem = "appletree",
		schem = "appletree",
		schem = "appletree",
		schem = "appletree",
		schem = "appletree",
		schem = "appletree",
		schem = "appletree",
		schem = "appletree",
		schem = "appletree",
		schem = "appletree",
		schem = "aspentree",
		schem = "aspentree",
		schem = "aspentree",
		schem = "aspentree",
		schem = "aspentree",
		schem = "aspentree",
		schem = "aspentree",
		schem = "aspentree",
		schem = "aspentree",
		schem = "aspentree",
		schem = "pinetree",
		schem = "pinetree",
		schem = "pinetree",
		schem = "pinetree",
		schem = "pinetree",
		schem = "pinetree",
		schem = "pinetree",
		schem = "pinetree",
		schem = "pinetree",
		schem = "pinetree",
		schem = "oldpinetree"
		}

```

Do you really want this? I don't

**Comment by paramat** [comment]
Yes it might be ok for small changes in chance but it isn't for anything else.

**Comment by MarkuBu** [comment]
Any progress on this? I would really like to see this feature in Minetest soon
